Code/Tools/ProjectManager/Source/ProjectUtils.cpp

```diff
@@ -507,5 +507,29 @@ namespace O3DE::ProjectManager
 
             return ProjectManagerScreen::Invalid;
         }
+
+        AZ::Outcome<QString, QString> ExecuteCommandResult(const QString& cmd, const QStringList& arguments, const QProcessEnvironment& processEnv)
+        {
+            QProcess execProcess;
+            execProcess.setProcessEnvironment(processEnv);
+            execProcess.setProcessChannelMode(QProcess::MergedChannels);
+            execProcess.start(cmd, arguments);
+            if (!execProcess.waitForStarted())
+            {
+                return AZ::Failure(QObject::tr("Unable to start process for command '%1'").arg(cmd));
+            }
+            constexpr int execTimeoutSecs = 30;
+            if (!execProcess.waitForFinished(execTimeoutSecs * 1000))
+            {
+                return AZ::Failure(QObject::tr("Process for command '%1' timed out at %2 seconds").arg(cmd).arg(execTimeoutSecs));
+            }
```

Comment:
I think this is better suited to be a parameter of this function.

The API doesn't document that the command has to finish in 30 seconds, so I believe it is better to allow the user to choose the timeout time

Reply:
The default argument for 'waitForFinished' is 30 seconds, i pulled it out just so I can give more information on an error. I can move the timeout value to the project defs instead.